import-scripts/import-msk-extract-projects.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+echo $(date)
+
+source $PORTAL_HOME/scripts/dmp-import-vars-functions.sh
+source $PORTAL_HOME/scripts/set-data-source-environment-vars.sh
+
+if [ -z $JAVA_BINARY ] | [ -z $GIT_BINARY ] | [ -z $PORTAL_HOME ] ; then
+    message="import-msk-extract-projects.sh cannot be run without setting JAVA_BINARY, GIT_BINARY, PORTAL_HOME...)"
+    echo $message
+    echo -e "$message" |  mail -s "update-msk-spectrum-cohort failed to run." $PIPELINES_EMAIL_LIST
+    sendPreImportFailureMessageMskPipelineLogsSlack "$message"
+    exit 2
+fi
+
+tmp=$PORTAL_HOME/tmp/import-msk-extract-projects
+if ! [ -d "$tmp" ] ; then
+    if ! mkdir -p "$tmp" ; then
+        echo "Error : could not create tmp directory '$tmp'" >&2
+        exit 1
+    fi
+fi
+if [[ -d "$tmp" && "$tmp" != "/" ]]; then
+    rm -rf "$tmp"/*
+fi
+
+source $PORTAL_HOME/scripts/clear-persistence-cache-shell-functions.sh
+
+IMPORT_FAIL=0
+extract_projects_notification_file=$(mktemp $tmp/import-msk-extract-projects-notification.$now.XXXXXX)
+# grab data from extract-projects repos
+fetch_updates_in_data_sources "extract-projects"
+
+# import data into msk portal
+$JAVA_BINARY -Xmx64g $JAVA_IMPORTER_ARGS --update-study-data --portal extract-projects-to-msk-portal --notification-file $extract_projects_notification_file --oncotree-version $ONCOTREE_VERSION_TO_USE --transcript-overrides-source mskcc --disable-redcap-export
+if [ $? -gt 0 ]; then
+    echo "MSK Extract projects import failed!"
+    IMPORT_FAIL=1
+    EMAIL_BODY="Extract projects import failed"
+    echo -e "Sending email $EMAIL_BODY"
+    echo -e "$EMAIL_BODY" | mail -s "Update failure: MSK Extract Projects" $PIPELINES_EMAIL_LIST
+fi
+
+# get num studies updated
+if [[ $? -eq 0 && -f "$TMP_DIRECTORY/num_studies_updated.txt" ]]; then
```

Comment:
The first "$?" in this condition test will not have the exit status code from the attempted import on line 35. Instead it will have the exit stats of the "if" statement from 36 though 42.

If the intention is default num_studies_updated to zero any time the import exit status is an error, then I suggest using the IMPORT_FAIL variable which would be 0 if the import did not fail, and 1 if it did fail. So perhaps:
if [[ $IMPORT_FAIL -eq 0 && -f "$TMP_DIRECTORY/num_studies_updated.txt" ]]; then